### ISSUE TITLE
资源和接口关联后怎么用的？

### ISSUE THREAD

**Issue by wjkang** [Jiiiiiin/jiiiiiin-security#4]
目前遇到了不同菜单下的页面会调用同一个接口的情况，比如在用户管理页面关联角色的时候需要调用角色的相关接口。而对角色授权的时候是按资源来的，比如“用户管理”这个菜单下的有个“关联角色”的按钮功能，授权了这个权限后，其实还要到“角色管理”菜单下授权相应的权限，因为做关联角色操作的时候需要调用查询角色的接口。资源和接口关联是为了处理这种问题吗

**Comment by Jiiiiiin** [comment]
对 就是因为会存在 一个资源可能会存在调用多个接口的情况，才又添加了接口资源关联表，这样做的好处就是资源只需要两种类型【菜单、按钮】，菜单就对应点击之后跳下去的页面初始化时候需要访问的后端接口，这个时候可能需要访问多个接口，那么单单用资源表一个字段来标识一组RESTful接口就很麻烦了

**Comment by Jiiiiiin** [comment]
另外其实这一层就把接口和角色分离开了，两则没有关系，角色只关联资源，再由资源去管理接口。

**Comment by wjkang** [comment]
嗯，明白，这样就完全动态的了。我之前采用的还是像没有前后端分离的那样，每个资源标识一个权限码，后端接口也用权限码标识（直接写死在代码上，我用的c#，类似java的注解），执行接口方法之前校验一下用具备的权限码是否包含当前接口方法标识的权限码。做资源和接口感觉会更加灵活，虽然麻烦点，但是却能把整个校验从接口程序中分离出来，毕竟拿到接口url就可以做校验了，不需要请求到达接口程序
